test/test_distributions.py

```diff
@@ -391,6 +399,24 @@ def get_sp_dist(jax_dist):
     T(dist.LogUniform, 1.0, 2.0),
     T(dist.LogUniform, 1.0, np.array([2.0, 3.0])),
     T(dist.LogUniform, np.array([1.0, 2.0]), np.array([[3.0], [4.0]])),
+    T(
+        dist.MatrixNormal,
+        1.0 * np.arange(6).reshape(3, 2),
+        np.array([[1.0, 0], [0.4, 1]]),
+        np.array([[1.0, 0, 0], [0.3, 0.36, 0], [0.4, 0.49, 4]]),
+    ),
+    T(
+        dist.MatrixNormal,
+        1.0 * np.arange(12).reshape((2, 3, 2)),
+        np.array([[1.0, 0], [0.4, 0.5]]) * np.ones((2, 2, 2)),
+        np.array([[1.0, 0, 0], [0.3, 0.36, 0], [0.4, 0.49, 4]]) * np.ones((2, 3, 3)),
+    ),
+    T(
+        dist.MatrixNormal,
+        1.0 * np.arange(36).reshape((2, 3, 3, 2)),
+        np.broadcast_to(np.identity(2), (2, 3, 2, 2)),
+        np.broadcast_to(np.identity(3), (2, 3, 3, 3)),
```

Comment:
Maybe we can remove this broadcast operator to check if our batching logic is working?

Reply:
Kepp `loc`as is and `np.identity(2)`, `np.identity(3)`?

Reply:
That does work but breaks the test since I'm looping over the batch dimension which do not match.
`samples.shape=sample_shape+batch_shape+event_shape` while `jax_dist.scale_tril_row=(1,1)+event_shape`.

Reply:
Found a solution.